src/scripts/main.js

```diff
@@ -1 +1,79 @@
 'use strict';
+
+const body = document.querySelector('body');
+
+const success = 'success';
+const error = 'error';
+
+const firstPromise = new Promise((resolve, reject) => {
+  document.addEventListener('click', function handleClick() {
+    resolve('First promise was resolved');
+    document.removeEventListener('click', handleClick);
+  });
+
+  setTimeout(() => {
+    reject(new Error('First promise was rejected'));
+  }, 3000);
+});
+
+firstPromise
+  .then((message) => {
+    createNotification(success, message);
+  })
+  .catch((errorMsg) => {
+    createNotification(error, errorMsg);
+  });
+
+function createNotification(type, message) {
+  const notification = document.createElement('div');
+
+  notification.setAttribute('data-qa', 'notification');
+  notification.classList.add(type);
+  notification.innerText = message;
+  body.appendChild(notification);
+}
+
+const secondPromise = new Promise((resolve) => {
+  document.addEventListener('click', function handleClick() {
+    resolve('Second promise was resolved');
+    document.removeEventListener('click', handleClick);
+  });
+
+  document.addEventListener('contextmenu', function handleClick(e) {
+    e.preventDefault();
+    resolve('Second promise was resolved');
+    document.removeEventListener('contextmenu', handleClick);
+  });
+});
```

Comment:
The second promise resolves on both click and contextmenu events, but it doesn't handle the case where both events might occur. This could lead to multiple resolutions of the promise, which is not allowed. Consider adding a flag to ensure the promise resolves only once.